features/gui/tasklist.lua

```diff
@@ -210,6 +210,13 @@ local function update_volunteer_button(button, task)
     end
 end
 
+local function update_top_gui(player)
+    local button = player.gui.top[main_button_name]
+    if button and button.valid then
+        button.number = table_size(tasks) or 0
```

Comment:
```suggestion
        button.number = #tasks
```
We can use the `#` operator in this case as the keys are consecutive. We use it below too. So for consistency and because it should perform better we should use it.

Reply:
Actually, the `table_size` methods is provided by the Factorio lib and implemented in C++, should be safer than the # operator

Reply:
`table_size` is O(n) whereas `#` is O(1). The reason for `table_size` is you can't use `#` when you use the table as a dict, but in this case we use it as an array.

Reply:
Alright, I will change it to # then, thanks for the clarification. I guess table_size is just the c++ interpretation of the Lua counterpart but not faster than # then 😄